opensquirrel/ir.py

```diff
@@ -670,6 +690,9 @@ def add_measure(self, measure: Measure) -> None:
     def add_reset(self, reset: Reset) -> None:
         self.statements.append(reset)
 
+    def add_statement(self, statement: Statement) -> None:
+        self.statements.append(statement)
```

Comment:
Why is this added? Is this for the gate modifiers? Shouldn't `add_gate` work for that?

Reply:
This is helpful for the refactoring of `parsing.py`. We can use an `add_statement` line to add gates and measure, and reset  instructions.